src/mjolnir/graphbuilder.cc

```diff
@@ -186,11 +187,11 @@ void ConstructEdges(const OSMData& osmdata,
         if (!infer_turn_channels && way.turn_channel())
           edge.attributes.turn_channel = true;
 
-        uint32_t size = static_cast<uint32_t>(edges.size());
-        if (!edge.attributes.way_begin)
-          size += 1;
+        // remember what edge this node will end, its complicated by the fact that we delay adding the
+        // edge until the next iteration of the loop, ie once the edge becomes prev_edge
+        uint32_t end_of = static_cast<uint32_t>(edges.size() + prev_edge.is_valid());
```

Comment:
i tried to make this easier to understand. let me know if the comment makes sense